File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/diary/editdiary/DatePickerDialogFragment.java
package com.websarva.wings.android.zuboradiary.ui.diary.editdiary;


import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.widget.DatePicker;

import androidx.fragment.app.DialogFragment;
import androidx.lifecycle.SavedStateHandle;
import androidx.navigation.NavBackStackEntry;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;

import java.time.LocalDate;
import java.util.Calendar;

public class DatePickerDialogFragment extends DialogFragment{
    private static final String fromClassName = "From" + DatePickerDialogFragment.class.getName();
    public static final String KEY_SELECTED_DATE = "SelectedDate" + fromClassName;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        LocalDate currentDate =
                DatePickerDialogFragmentArgs.fromBundle(requireArguments()).getCurrentDate();
        return new DatePickerDialog(
                requireContext(),
                new DatePickerDialog.OnDateSetListener() {
                    @Override
                    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
                        LocalDate selectedDate = LocalDate.of(year, month + 1, dayOfMonth);
                        NavController navController =
                                NavHostFragment.findNavController(
                                        DatePickerDialogFragment.this);
                        NavBackStackEntry navBackStackEntry = navController.getPreviousBackStackEntry();
                        if (navBackStackEntry != null) {
                            SavedStateHandle savedStateHandle = navBackStackEntry.getSavedStateHandle();
                            savedStateHandle.set(KEY_SELECTED_DATE, selectedDate);
                        }
                    }
                },
                currentDate.getYear(),
                currentDate.getMonthValue() - 1,
                currentDate.getDayOfMonth()
        );
    }
}

File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/fragment/dialog/DatePickerDialogFragment.kt
package com.websarva.wings.android.zuboradiary.ui.fragment.dialog

import java.time.LocalDate


class DatePickerDialogFragment : BaseDatePickerDialogFragment() {

    companion object {
        private val fromClassName = "From" + DatePickerDialogFragment::class.java.name
        @JvmField
        val KEY_SELECTED_DATE: String = "SelectedDate$fromClassName"
    }

    override fun createInitialDate(): LocalDate {
        return DatePickerDialogFragmentArgs.fromBundle(requireArguments()).date
    }

    override fun handleOnPositiveButtonClick(selectedDate: LocalDate) {
        setSelectedDate(selectedDate)
    }

    override fun handleOnNegativeButtonClick() {
        setSelectedDate(null)
    }

    override fun handleOnCancel() {
        setSelectedDate(null)
    }

    private fun setSelectedDate(selectedDate: LocalDate?) {
        setResult(KEY_SELECTED_DATE, selectedDate)
    }
}
